e2e_tests/scenarios/tests/common/docker_compose.rs

```diff
@@ -0,0 +1,141 @@
+use std::process::{Command, Stdio};
+use std::time::{Duration, Instant};
+
+use anyhow::{Context, Error};
+use hyper::{Body, Client};
+use tracing::{debug, info};
+use tracing_subscriber::fmt::format;
+
+use crate::common::prelude::{Request, StatusCode};
+
+const LOCAL_ADDR: &str = "localhost:9080";
+
+pub struct DockerComposeGuard<'a> {
+    // Current working dir containing compose.yml
+    cwd: &'a str,
+}
+
+impl<'a> DockerComposeGuard<'a> {
+    pub fn get_local_addr(&self) -> &'a str {
+        LOCAL_ADDR
+    }
+
+    pub async fn restart_sequencer(&self) -> anyhow::Result<()> {
+        let (stdout, stderr) = run_cmd_to_output(
+            self.cwd,
+            "docker compose --profile e2e-ha restart signup-sequencer-0",
+        )
+        .context("Restarting sequencer.")?;
+
+        debug!(
+            "Docker compose rstart output:\n stdout:\n{}\nstderr:\n{}\n",
+            stdout, stderr
+        );
+
+        await_running(self).await
+    }
+}
+
+impl<'a> Drop for DockerComposeGuard<'a> {
+    fn drop(&mut self) {
+        // May run when compose is not up but better to be sure its down.
+        // Parameter '-v' is removing all volumes and networks.
+        if let Err(err) = run_cmd(
+            self.cwd,
+            &format!("docker compose --profile e2e-ha down -v"),
+        ) {
+            eprintln!("Failed to put down docker compose: {}", err);
+        }
+    }
+}
+
+/// Starts a docker compose infrastructure. It will be stopped and removed when
+/// the guard is dropped.
+///
+/// Note that we're using sync code here so we'll block the executor - but this
+/// is fine, because the spawned container will still run in the background.
+pub async fn setup(cwd: &str) -> anyhow::Result<DockerComposeGuard> {
+    let res = DockerComposeGuard { cwd };
+
+    let (stdout, stderr) = run_cmd_to_output(res.cwd, "docker compose --profile e2e-ha up -d")
```

Comment:
In order to run multiple envs in parallel we would want to run with `--project-name {random_generated_id_or_string}`